products/bigquery/api.yaml

```diff
@@ -1433,3 +1433,139 @@ objects:
                         - :FLOAT
                         - :BOOLEAN
+  - !ruby/object:Api::Resource
+    name: 'Routine'
+    kind: 'bigquery#routine'
+    base_url: projects/{{project}}/datasets/{{dataset_id}}/routines
+    self_link: projects/{{project}}/datasets/{{dataset_id}}/routines/{{routine_id}}
+    description: |
+      A user-defined function or a stored procedure that belongs to a Dataset
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Routines Intro': 'https://cloud.google.com/bigquery/docs/reference/rest/v2/routines'
+      api: 'https://cloud.google.com/bigquery/docs/reference/rest/v2/routines'
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: routineReference
+        description: Reference describing the ID of this routine
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'datasetId'
+            description: The ID of the dataset containing this routine
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'projectId'
+            description: The ID of the project containing this routine
+          - !ruby/object:Api::Type::String
+            name: 'routineId'
+            description: The ID of the the routine. The ID must contain only
+              letters (a-z, A-Z), numbers (0-9), or underscores (_).
+              The maximum length is 256 characters.
+            required: true
+      - !ruby/object:Api::Type::Enum
+        name: 'routineType'
+        description: The type of routine.
+        values:
+          - :ROUTINE_TYPE_UNSPECIFIED
+          - :SCALAR_FUNCTION
+          - :PROCEDURE
+      - !ruby/object:Api::Type::Integer
+        name: 'creationTime'
+        output: true
+        description: |
+          The time when this routine was created, in milliseconds since the
+          epoch.
+      - !ruby/object:Api::Type::Integer
+        name: 'lastModifiedTime'
+        output: true
+        description: |
+          The time when this routine was modified, in milliseconds since the
+          epoch.
+      - !ruby/object:Api::Type::Enum
+        name: 'language'
+        description: |
+          The language of the routine.
+        values:
+          - :LANGUAGE_UNSPECIFIED
+          - :SQL
+          - :JAVASCRIPT
+      - !ruby/object:Api::Type::Array
+        name: 'arguments'
+        description: Input/output argument of a function or a stored procedure.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'name'
+              description: |
+                The name of this argument. Can be absent for function return argument.
+            - !ruby/object:Api::Type::Enum
+              name: 'argumentKind'
+              description: Defaults to FIXED_TYPE.
+              values:
+                - :ARGUMENT_KIND_UNSPECIFIED
```

Comment:
(same drill as above re: unspecified for the rest of these enums)

Reply:
removed